mod_elasticsearch2.erl

```diff
@@ -55,18 +64,121 @@
     terminate/2,
     code_change/3,
 
-    prepare_index/1
+    prepare_index/1,
+    delete_recreate_index/1
 ]).
 
 -include_lib("zotonic.hrl").
 -include("include/elasticsearch2.hrl").
 
--record(state, {context}).
+% Period during which update commands are buffered.
+-define(BULK_DELAY, 1000).
+
+% Max number of batched commands before flush is forced.
+-define(BULK_MAX_BATCH, 500).
+
+
+-record(state, {
+        bulk = [] :: list( elasticsearch2:bulkcmd() ),
+        timer = undefined :: timer:tref() | undefined,
+        bulk_pid = undefined :: undefined | pid(),
+        context :: z:context(),
+        total = 0 :: non_neg_integer(),
```

Comment:
Als ik dit zo lees, zou ik verwachten dat deze `total` gelijk is aan `total_ok + total_error`, maar dat is niet altijd zo, omdat `total` al wordt opgehoogd wanneer er een command gequeued wordt en `total_ok` en `total_error` pas als de commands verwerkt zijn. Maar als je dat recht zou trekken door `total` ook pas te updaten wanneer commands verwerkt zijn is het wel altijd `total_ok + total_error`, dus dan heb je hem eigenlijk ook niet meer nodig.

Als je hem wel wil zouden zou ik er even een comment bij zetten met dit verschil.

Reply:
`total` is inderdaad `total_received`.  Dat zou dan `total_ok` + `total_error` + `queued` moeten zijn, als er niet ergens iets verloren is gegaan.

Reply:
Done, is now called `total_received`